// @jc21/cypress-jsonschema-validation/lib/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const chalk = require("chalk");
const _ = require("lodash");
class Logger {
    constructor(section) {
        this.title = section;
    }
    success(...args) {
        const arr = _.values(args);
        arr.unshift(this.getTitle(chalk.green.bold('SUCCESS')));
        // @ts-ignore
        console.log.apply(null, arr);
    }
    error(...args) {
        const arr = _.values(args);
        arr.unshift(this.getTitle(chalk.red.bold('ERROR')));
        // @ts-ignore
        console.log.apply(null, arr);
    }
    info(...args) {
        const arr = _.values(args);
        arr.unshift(this.getTitle(chalk.blue.bold('INFO')));
        // @ts-ignore
        console.log.apply(null, arr);
    }
    debug(...args) {
        const arr = _.values(args);
        arr.unshift(this.getTitle(chalk.magenta.bold('DEBUG')));
        // @ts-ignore
        console.log.apply(null, arr);
    }
    getTitle(type) {
        return chalk.blue.bold('[') + chalk.cyan.bold(this.title) + ' ' + type + chalk.blue.bold(']');
    }
}
exports.default = Logger;
